src/arviz_stats/psense.py

```diff
@@ -0,0 +1,199 @@
+    """
+    dataset = extract(
+        dt, var_names=var_names, filter_vars=filter_vars, group="posterior", combined=False
+    )
+    if coords is not None:
+        dataset = dataset.sel(coords)
+
+    lower_alpha = 1 / (1 + delta)
+    upper_alpha = 1 + delta
+
+    lower_w = _get_power_scale_weights(
+        dt,
+        alpha=lower_alpha,
+        group=group,
+        sample_dims=sample_dims,
+        component_var_names=component_var_names,
+        component_coords=component_coords,
+    )
+
+    upper_w = _get_power_scale_weights(
+        dt,
+        alpha=upper_alpha,
+        group=group,
+        sample_dims=sample_dims,
+        component_var_names=component_var_names,
+        component_coords=component_coords,
+    )
+
+    return dataset.azstats.power_scale_sense(
+        lower_w=lower_w,
+        upper_w=upper_w,
+        delta=delta,
+        dims=sample_dims,
+    )
+
+
+def psense_summary(data, threshold=0.05, round_to=3):
+    """
+    Compute the prior/likelihood sensitivity based on power-scaling perturbations.
+
+    Parameters
+    ----------
+    data : DataTree
+    threshold : float, optional
+        Threshold value to determine the sensitivity diagnosis. Default is 0.05.
+    round_to : int, optional
+        Number of decimal places to round the sensitivity values. Default is 3.
+
+    Returns
+    -------
+    psense_df : DataFrame
+        DataFrame containing the prior and likelihood sensitivity values for each variable
+        in the data. And a diagnosis column with the following values:
+        - "prior-data conflict" if both prior and likelihood sensitivity are above threshold
+        - "strong prior / weak likelihood" if the prior sensitivity is above threshold
+        and the likelihood sensitivity is below the threshold
+        - "-" otherwise
+    """
+    pssdp = psense(data, group="log_prior")
+    pssdl = psense(data, group="log_likelihood")
+
+    joined = xr.concat([pssdp, pssdl], dim="component").assign_coords(
+        component=["prior", "likelihood"]
+    )
+
+    n_vars = np.sum([joined[var].size // 2 for var in joined.data_vars])
+
+    psense_df = pd.DataFrame(
+        (np.full((cast(int, n_vars), 2), np.nan)), columns=["prior", "likelihood"]
+    )
+
+    indices = []
+    for i, (var_name, sel, isel, values) in enumerate(
+        xarray_var_iter(joined, skip_dims={"component"})
+    ):
+        psense_df.iloc[i] = values
+        indices.append(labeller.make_label_flat(var_name, sel, isel))
+    psense_df.index = indices
+
+    def _diagnose(row):
+        if row["prior"] >= threshold and row["likelihood"] >= threshold:
+            return "prior-data conflict"
+        if row["prior"] > threshold > row["likelihood"]:
+            return "strong prior / weak likelihood"
+
+        return "-"
+
+    psense_df["diagnosis"] = psense_df.apply(_diagnose, axis=1)
+
+    return psense_df.round(round_to)
+
+
+def _get_power_scale_weights(
```

Comment:
I prefer the original version. Or maybe have this function return both lower_w and upper_w?

Subsetting and summing over the "latent-obs_var" dimension should generally not be expensive computations, but they are not free and there is also no need to do that twice.

Reply:
Only one of them? or both of them? If both we can modify the function to return both lower_w and upper_w. Otherwise it is probably better to have this be a `get_component_draws` function so only 2 lines are needed after that to get the weights